openvasd/openvasd.c

```diff
@@ -123,69 +100,41 @@ struct openvasd_vt_single
   GHashTable *vt_values;
 };
 
-/**
- * @brief Request methods
- */
-enum openvas_request_method
-{
-  POST,
-  GET,
-  HEAD,
-  DELETE,
-};
-
-typedef enum openvas_request_method openvasd_req_method_t;
-
 /**
  * @brief Allocate openvasd curl handler
  *
- * @return Openvasd curl handler.
+ * @return openvasd curl handler.
  */
-static openvasd_curlm_t *
+static curlutils_multi_t *
 openvasd_curlm_handler_new (void)
 {
-  return (openvasd_curlm_t *) g_malloc0 (sizeof (struct openvasd_curlm));
+  return (curlutils_multi_t *) g_malloc0 (sizeof (struct curlutils_multi));
 }
 
 /**
  * @brief Cleanup an openvasd curl handler
  *
- * @param h Openvasd curl handler to clean
+ * @param h openvasd curl handler to clean
  */
 static void
-openvasd_curlm_handler_close (openvasd_curlm_t *h)
+openvasd_curlm_handler_close (curlutils_multi_t *h)
 {
-  int queued = 0;
-
-  /* when an easy handle has completed, remove it */
-  CURLMsg *msg = curl_multi_info_read (h->h, &queued);
-  if (msg)
-    {
-      if (msg->msg == CURLMSG_DONE)
-        {
-          curl_multi_remove_handle (h->h, msg->easy_handle);
-          curl_easy_cleanup (msg->easy_handle);
-          curl_slist_free_all (h->customheader);
-          curl_multi_cleanup (h->h);
-          return;
-        }
-      g_warning ("%s: Not possible to clean up the curl handler", __func__);
-    }
+  curlutils_multi_cleanup (h);
 }
 
 /** @brief Allocate the vt stream struct to hold the response
  *  and the curlm handler
  *
  *  @return The vt stream struct. Must be free with openvasd_vt_stream_new().
  */
-static openvasd_vt_stream_t
+static curlutils_response_stream_t
 openvasd_vt_stream_new (void)
```

Comment:
This is all about allocating a `curlutils_response_stream_t`, so it would be better as `curlutils_response_stream_new` in curlutils.

Reply:
Same for the free and reset.